datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -49,16 +50,17 @@ def compute_derivative_rows(self, rows, metrics, key):
             prev = self.previous_statements.get(row_key)
             if prev is None:
                 continue
+
             metric_columns = metrics & set(row.keys())
             dropped_metrics.update(metrics - metric_columns)
+
             if any([row[k] - prev[k] < 0 for k in metric_columns]):
                 # The table was truncated or stats reset; begin tracking again from this point
                 continue
             if all([row[k] - prev[k] == 0 for k in metric_columns]):
                 # No metrics to report; query did not run
                 continue
-            result.append({k: row[k] - prev[k] if k in metric_columns else row[k]
-                           for k in row})
+            result.append({k: row[k] - prev[k] if k in metric_columns else row[k] for k in row.keys()})
```

Comment:
Aren't these functionally the same?

Reply:
No, there was a weird bug. `k in row` vs `k in row.keys()`. The postgres one was behaving like an array and causing key error. 🤷